airbyte-lib/airbyte_lib/datasets/_map.py

```diff
@@ -0,0 +1,23 @@
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+
+"""A generic interface for a set of streams."""
+
+from collections.abc import Iterator, Mapping
+
+from airbyte_lib.datasets._base import DatasetBase
+
+
+class DatasetMap(Mapping):
```

Comment:
This doesn't seem to be used, is it meant to form the basis of ReadResult? (related to my other comment about overriding `items()` on the read result, I guess this would be the way to do it.

Reply:
Yep. This is still only half baked, but that's the goal.